src/main/java/com/epam/aidial/core/token/TokenUsage.java

```diff
@@ -1,12 +1,49 @@
 package com.epam.aidial.core.token;
 
+import com.epam.aidial.core.config.Pricing;
 import lombok.Data;
 
 @Data
 public class TokenUsage {
     private long completionTokens;
     private long promptTokens;
     private long totalTokens;
+    private Double cost;
+    private Double aggCost;
+
+    public void calculateCost(Pricing pricing) {
+        if (pricing == null) {
+            return;
+        }
+        String unit = pricing.getUnit();
+        if (!"token".equals(unit)) {
+            return;
+        }
+        double cost = 0.0;
+        if (pricing.getPrompt() != null) {
+            cost += promptTokens * Double.parseDouble(pricing.getPrompt());
```

Comment:
Can throw an exception, so the calling code should take care of this :(

Reply:
fixed